woocommerce/woocommerce-gutenberg-products-block
# Try namespacing callbacks added by extensions for `canMakePayment`

This PR won't be merged, it's an experiment to see how adding `cart` to `canPayArgument` affects things, as well as changing the API for how extensions will register their own callbacks to `canMakePayment`.

Cc @ralucaStan 

### How to test the changes in this Pull Request:

1. Add this code somewhere in an extension
2. 
```js
registerPaymentMethodExtensionCallbacks( 'woocommerce-marketplace-extension',
	{
		stripe: ( arg ) => { console.log("callback on woocommerce-marketplace-extension called", arg); throw new Error('Some error thrown by the extension'); return true; },
		cod: ( arg ) => { /* code */ return true; },
		cheque: ( arg ) => { /* code */ return true; }
	}
);
```
2. Check the `stripe` payment method doesn't register, and check the console for an error.
3. Remove the `throw new Error` part of the callback, and check stripe registers.
4. Make any of the callbacks return false and check they don't register.



### Changelog

> Add suggested changelog entry here.
